test/tests/auxkernels/lower_d_var/tests

```diff
@@ -0,0 +1,10 @@
+[Tests]
+  design = 'AuxKernels/index.md'
+  issues = '#26592'
+  [csv]
+    type = CSVDiff
+    input = lower-d-aux-var.i
+    csvdiff = lower-d-aux-var_out.csv
+    requirement = 'The system shall allow populating lower-dimensional auxiliary variables using boundary restricted auxiliary kernels.'
+  []
+[]
```

Comment:
can you please add a h-refined test case?
That would be my first worry on such changes

Reply:
If someone just used `RefineSidesetGenerator`, then we won't have matching refinement of the lower-d elements and the higher-d boundary elements. So your worry is valid ... the code being added here and likely a lot of lower-d element code in MOOSE will fail with refinement

Reply:
never mind, I was wrong 😄 we can test this